dnapitest/dnapitest.go

```diff
@@ -67,6 +67,14 @@ func (s *Server) handler(w http.ResponseWriter, r *http.Request) {
 		s.handlerEnroll(w, r)
 	case message.EndpointV1:
 		s.handlerDNClient(w, r)
+	case message.PreAuthEndpoint:
+		expected := s.expectedRequests[0]
+		s.expectedRequests = s.expectedRequests[1:]
+		res := expected.dncRequestResponse
+		w.WriteHeader(res.statusCode)
+		_, _ = w.Write(res.response(message.RequestWrapper{}))
```

Comment:
This does not look correct to me. This is for a /v1/dnclient endpoint message.

Doesn't preauth need to return a pollToken in order for the poll endpoint to behave correctly? I think we should add a test that the handler receives the poll token sent by the client.

Reply:
It does! The test injects the "desired response" from the server, and confirms the client gets it. There's no "real handler" because PreAuth doesn't take arguments from the client.